File: build_wp_theme.js
const fs = require('fs');
const path = require('path');

const themeDir = 'resurs11-theme';
if (!fs.existsSync(themeDir)) {
    fs.mkdirSync(themeDir);
}

fs.writeFileSync(path.join(themeDir, 'style.css'), '/*\nTheme Name: Resurs 11\nAuthor: AI Assistant\nDescription: Custom theme for Resurs 11\nVersion: 1.0\n*/\n', 'utf8');
fs.writeFileSync(path.join(themeDir, 'index.php'), '<?php get_header(); ?>\n<?php get_footer(); ?>\n', 'utf8');

const indexHtml = fs.readFileSync('index.html', 'utf8');

const headerMatch = indexHtml.match(/(.*?<\/header>)/is);
let headerContent = headerMatch ? headerMatch[1] : '';

const footerMatch = indexHtml.match(/(<footer\b.*)/is);
let footerContent = footerMatch ? footerMatch[1] : '';

headerContent = headerContent.replace(/(<\/head>)/i, '<?php wp_head(); ?>\n$1');
footerContent = footerContent.replace(/(<\/body>)/i, '<?php wp_footer(); ?>\n$1');

function updateImages(text) {
    if (!text) return text;
    let res = text.replace(/src=["']images\//ig, 'src="<?php echo get_template_directory_uri(); ?>/images/');
    res = res.replace(/onerror=["']this\.src=['"]images\//ig, 'onerror="this.src=\'<?php echo get_template_directory_uri(); ?>/images/');
    return res;
}

function updateLinks(text) {
    if (!text) return text;
    let res = text.replace(/href=["']index\.html["']/gi, 'href="<?php echo home_url(); ?>"');
    res = res.replace(/href=["']([a-zA-Z0-9_\-]+)\.html["']/gi, 'href="<?php echo home_url(\'/$1/\'); ?>"');
    return res;
}

headerContent = updateImages(headerContent);
headerContent = updateLinks(headerContent);
footerContent = updateImages(footerContent);
footerContent = updateLinks(footerContent);

fs.writeFileSync(path.join(themeDir, 'header.php'), headerContent, 'utf8');
fs.writeFileSync(path.join(themeDir, 'footer.php'), footerContent, 'utf8');

function extractBody(html) {
    const match = html.match(/<\/header>(.*?)<footer\b/is);
    return match ? match[1].trim() : '';
}

let frontBody = extractBody(indexHtml);
frontBody = updateImages(frontBody);
frontBody = updateLinks(frontBody);
fs.writeFileSync(path.join(themeDir, 'front-page.php'), '<?php get_header(); ?>\n' + frontBody + '\n<?php get_footer(); ?>\n', 'utf8');

const files = fs.readdirSync('.');
const skipFiles = ['index.html', 'temp_peleti.html', 'peleti-sun.html', 'eko_page.html', 'peleti_page.html'];

files.forEach(file => {
    if (file.endsWith('.html') && !skipFiles.includes(file)) {
        const html = fs.readFileSync(file, 'utf8');
        let bodyContent = extractBody(html);
        bodyContent = updateImages(bodyContent);
        bodyContent = updateLinks(bodyContent);
        
        const templateName = file.replace('.html', '').replace(/-/g, ' ').replace(/\b\w/g, l => l.toUpperCase());
        const phpFilename = 'page-' + file.replace('.html', '.php');
        
        fs.writeFileSync(path.join(themeDir, phpFilename), '<?php\n/* Template Name: ' + templateName + ' */\nget_header();\n?>\n' + bodyContent + '\n<?php get_footer(); ?>\n', 'utf8');
    }
});

console.log('Theme generated successfully in resurs11-theme folder.');

File: build_wp_theme.py
import os
import re

theme_dir = 'resurs11-theme'
os.makedirs(theme_dir, exist_ok=True)

with open(os.path.join(theme_dir, 'style.css'), 'w', encoding='utf-8') as f:
    f.write('/*\nTheme Name: Resurs 11\nAuthor: AI Assistant\nDescription: Custom theme for Resurs 11\nVersion: 1.0\n*/\n')

with open(os.path.join(theme_dir, 'index.php'), 'w', encoding='utf-8') as f:
    f.write('<?php get_header(); ?>\n<?php get_footer(); ?>\n')

with open('index.html', 'r', encoding='utf-8') as f:
    index_html = f.read()

header_match = re.search(r'(.*?</header>)', index_html, re.DOTALL | re.IGNORECASE)
header_content = header_match.group(1) if header_match else ''

footer_match = re.search(r'(<footer\b.*)', index_html, re.DOTALL | re.IGNORECASE)
footer_content = footer_match.group(1) if footer_match else ''

header_content = re.sub(r'(</head>)', r'<?php wp_head(); ?>\n\1', header_content, flags=re.IGNORECASE)
footer_content = re.sub(r'(</body>)', r'<?php wp_footer(); ?>\n\1', footer_content, flags=re.IGNORECASE)

def update_images(text):
    text = re.sub(r'src=["\']images/', r'src="<?php echo get_template_directory_uri(); ?>/images/', text)
    text = re.sub(r'onerror=["\']this\.src=[\'\"]images/', r'onerror="this.src=\'<?php echo get_template_directory_uri(); ?>/images/', text)
    return text

header_content = update_images(header_content)
footer_content = update_images(footer_content)

with open(os.path.join(theme_dir, 'header.php'), 'w', encoding='utf-8') as f:
    f.write(header_content)

with open(os.path.join(theme_dir, 'footer.php'), 'w', encoding='utf-8') as f:
    f.write(footer_content)

def extract_body(html):
    match = re.search(r'</header>(.*?)<footer\b', html, re.DOTALL | re.IGNORECASE)
    return match.group(1).strip() if match else ''

front_body = extract_body(index_html)
front_body = update_images(front_body)
with open(os.path.join(theme_dir, 'front-page.php'), 'w', encoding='utf-8') as f:
    f.write('<?php get_header(); ?>\n' + front_body + '\n<?php get_footer(); ?>\n')

html_files = [f for f in os.listdir('.') if f.endswith('.html') and f not in ('index.html', 'temp_peleti.html', 'peleti-sun.html', 'eko_page.html', 'peleti_page.html')]
for html_file in html_files:
    with open(html_file, 'r', encoding='utf-8') as f:
        html = f.read()
    
    body_content = extract_body(html)
    body_content = update_images(body_content)
    
    template_name = html_file.replace('.html', '').replace('-', ' ').title()
    
    # Just prefix with page-
    php_filename = 'page-' + html_file.replace('.html', '.php')
        
    with open(os.path.join(theme_dir, php_filename), 'w', encoding='utf-8') as f:
        f.write('<?php\n/* Template Name: ' + template_name + ' */\nget_header();\n?>\n')
        f.write(body_content)
        f.write('\n<?php get_footer(); ?>\n')

print("Theme generated successfully in resurs11-theme folder.")
